command/connect/redirecttraffic/redirect_traffic.go

```diff
@@ -132,23 +148,72 @@ func (c *cmd) generateConfigFromFlags() (iptables.Config, error) {
 			return iptables.Config{}, fmt.Errorf("service %s is not a proxy service", c.proxyID)
 		}
 
-		cfg.ProxyInboundPort = svc.Port
+		// Decode proxy's opaque config so that we can use it later to configure
+		// traffic redirection with iptables.
 		var trCfg trafficRedirectProxyConfig
 		if err := mapstructure.WeakDecode(svc.Proxy.Config, &trCfg); err != nil {
 			return iptables.Config{}, fmt.Errorf("failed parsing Proxy.Config: %s", err)
 		}
 
+		// Set the proxy's inbound port.
+		cfg.ProxyInboundPort = svc.Port
 		if trCfg.BindPort != 0 {
 			cfg.ProxyInboundPort = trCfg.BindPort
 		}
 
-		// todo: Change once it's configurable
+		// Set the proxy's outbound port.
 		cfg.ProxyOutboundPort = iptables.DefaultTProxyOutboundPort
+		if svc.Proxy.TransparentProxy.OutboundListenerPort != 0 {
```

Comment:
TransparentProxy should have a nil check here since it's a pointer. 

Consul currently always returns a non-nil struct even if it's empty, but that's likely going to change for GA.

Reply:
Great catch! let me update it.